### PR TITLE
1586 netcdf meta

### PR DESCRIPTION
**Pull request by gantian127**

### CODE REVIEW

**Comment by pkdash** [comment]
@gantian127 Seems this PR has some conflicts that you need to resolve. I assume this work is not needed for AGU release.

File: hs_app_netCDF/forms.py [hunk 1]
```diff
@@ -1,67 +1,77 @@
...
         self.allow_edit = allow_edit
 
     @property
     def form_id(self):
-        form_id = 'id_original_coverage_%s'% self.number
+        form_id = 'id_original_coverage_%s' % self.number
         return form_id
 
     @property
     def form_if_button(self):
```
Comment (pkdash) [inline hs_app_netCDF/forms.py]: I think this property name supposed to be ```form_id_button```, right?

File: hs_app_netCDF/forms.py [hunk 2]
```diff
@@ -140,10 +156,11 @@ def __init__(self, *args, **kwargs):
                      Field('descriptive_name', css_class=field_width),
                      Field('method', css_class=field_width),
                      Field('missing_value', css_class=field_width)
-                ),
+                     ),
             )
 
-       # super(VariableFormHelper, self).__init__(allow_edit, res_short_id,  element_id, element_name, layout,  *args, **kwargs)
+        # super(VariableFormHelper, self).__init__(allow_edit, res_short_id,
```
Comment (pkdash) [inline hs_app_netCDF/forms.py]: Better to remove these commented code.

File: hs_app_netCDF/nc_functions/nc_dump.py [hunk 3]
```diff
@@ -36,14 +41,15 @@ def get_netcdf_header_file(nc_file_name, dump_folder=''):
     """
 
     # create a new text file
-    nc_file_basename = '.'.join(basename(nc_file_name).split('.')[:-1]) # name with no file extension
+    # name with no file extension
+    nc_file_basename = '.'.join(basename(nc_file_name).split('.')[:-1])
     nc_dump_file_folder = dump_folder if dump_folder else os.getcwd()
     nc_dump_file_name = nc_dump_file_folder+'/'+nc_file_basename+'_header_info.txt'
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_dump.py]: Add a space around the ```+``` operator.

File: hs_app_netCDF/nc_functions/nc_dump.py [hunk 4]
```diff
@@ -36,14 +41,15 @@ def get_netcdf_header_file(nc_file_name, dump_folder=''):
...
     nc_dump_file = open(nc_dump_file_name, 'w')
 
     # write the nc_dump string in text fle
-    dump_string = get_nc_dump_string_by_ncdump(nc_file_name) if get_nc_dump_string_by_ncdump(nc_file_name) \
-        else get_nc_dump_string(nc_file_name)
+    dump_string = get_nc_dump_string_by_ncdump(nc_file_name) \
+        if get_nc_dump_string_by_ncdump(nc_file_name) else get_nc_dump_string(nc_file_name)
     if dump_string:
         nc_dump_file.write(dump_string)
         print '{0}_header_info.txt is created'.format(nc_file_basename)
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_dump.py]: I would rather use logging instead of ```print```

File: hs_app_netCDF/nc_functions/nc_meta.py [hunk 5]
```diff
@@ -51,7 +59,7 @@ def get_nc_meta_dict(nc_file_name):
     return nc_meta_dict
 
 
-# Functions for dublin  core meta #####################################################################################
+# Functions for dublin  core meta ###############################################################
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_meta.py]: Delete this comment and add similar comment to the docstring of the function.

File: hs_app_netCDF/nc_functions/nc_meta.py [hunk 6]
```diff
@@ -410,15 +440,15 @@ def get_limit_meta_by_coor_type(nc_dataset, coor_type, coor_type_mapping):
     return limit_meta
 
 
-# Functions for type specific meta ##################################################################################
+# Functions for type specific meta ##############################################
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_meta.py]: Remove all these ```####### ....```

File: hs_app_netCDF/nc_functions/nc_utils.py [hunk 7]
```diff
@@ -1,20 +1,23 @@
...
 import re
 from collections import OrderedDict
+
+import osr
+import netCDF4
 import numpy
 
 
-# Functions for General Purpose ####################################################################################
+# Functions for General Purpose #################################################
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_utils.py]: As before, remove all these ```#### ...```

File: hs_app_netCDF/nc_functions/nc_utils.py [hunk 8]
```diff
@@ -182,12 +192,13 @@ def get_nc_variable_coordinate_meta(nc_dataset, nc_variable_name):
     return nc_variable_coordinate_meta
 
 
-# Functions for Coordinate Variable#####################################################################################
+# Functions for Coordinate Variable################################################
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_utils.py]: As before.

**Comment by pkdash** [comment]
@gantian127 Seems like some pylint cleanup is needed.

File: hs_app_netCDF/nc_functions/nc_meta.py [hunk 9]
```diff
@@ -59,7 +59,7 @@ def get_nc_meta_dict(nc_file_name):
     return nc_meta_dict
 
 
-# Functions for dublin  core meta ###############################################################
+# Functions for dublin core meta
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_meta.py]: Would be better to remove this comment here and if needed same comment can be included as part of the function docstring.

File: hs_app_netCDF/nc_functions/nc_meta.py [hunk 10]
```diff
@@ -440,7 +440,7 @@ def get_limit_meta_by_coor_type(nc_dataset, coor_type, coor_type_mapping):
     return limit_meta
 
 
-# Functions for type specific meta ##############################################
+# Functions for type specific meta
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_meta.py]: As above.

File: hs_app_netCDF/nc_functions/nc_utils.py [hunk 11]
```diff
@@ -17,7 +17,7 @@
 import numpy
 
 
-# Functions for General Purpose #################################################
+# Functions for General Purpose
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_utils.py]: As above

File: hs_app_netCDF/nc_functions/nc_utils.py [hunk 12]
```diff
@@ -71,7 +71,7 @@ def get_nc_variable_original_meta(nc_dataset, nc_variable_name):
     return nc_variable_original_meta
 
 
-# Functions for coordinate information of the dataset ########################################
+# Functions for coordinate information of the dataset
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_utils.py]: As above

File: hs_app_netCDF/nc_functions/nc_utils.py [hunk 13]
```diff
@@ -341,7 +341,7 @@ def get_nc_data_variable_namelist(nc_dataset):
     return nc_data_variable_namelist
 
 
-# Function for Grid Mapping Variable ########################################################
+# Function for Grid Mapping Variable
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_utils.py]: As above

File: hs_app_netCDF/nc_functions/nc_utils.py [hunk 14]
```diff
@@ -310,7 +310,7 @@ def get_nc_coordinate_bounds_variable_namelist(nc_dataset):
     return nc_coordinate_bounds_variable_namelist
 
 
-# Function for Data Variable ################################################
+# Function for Data Variable
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_utils.py]: As above

**Comment by pkdash** [comment]
@gantian127 I have now finished reviewing this PR with some comments.

File: hs_app_netCDF/nc_functions/nc_meta.py [hunk 15]
```diff
@@ -51,7 +59,7 @@ def get_nc_meta_dict(nc_file_name):
     return nc_meta_dict
 
 
-# Functions for dublin  core meta #####################################################################################
+# Functions for dublin core meta
```
Comment (pkdash) [inline hs_app_netCDF/nc_functions/nc_meta.py]: It seems this comment is for a group functions.  OK to leave as it is.

File: hs_app_netCDF/receivers.py [hunk 16]
```diff
@@ -168,39 +168,6 @@ def netcdf_pre_create_resource(sender, **kwargs):
             shutil.rmtree(os.path.dirname(in_file_name))
 
 
-# # receiver used to create netcdf header text after user click on "create resource"
```
Comment (pkdash) [inline hs_app_netCDF/receivers.py]: Glad to see this commented code removed.

**Comment by pkdash** [comment]
@gantian127 Is this ready fro final round of review?

**Comment by gantian127** [comment]
@pkdash If there is no need for model migration, the code should be fine for review. I have sent you an email about the model migration today. Would you please have a look at it and reply me? Thanks

**Comment by pkdash** [comment]
+1

**Comment by pkdash** [comment]
@gantian127 You can merge now.

**Pull request merged**